.ci/jenkins/Jenkinsfile.prod.nightly

```diff
@@ -217,4 +220,15 @@ String calculateArchiveRepoBranch() {
     }
     // e.g., main
     return branch
+}
+
+
+// Parse version from main branch of the given project
+//      * project: in the form of owner/repository
+def parseVersionFromPom(String project) {
+    def pomFilename = "${project.replaceAll("/", "_")}_pom.xml"
+
+    sh "curl https://raw.githubusercontent.com/${project}/main/pom.xml -o ${env.WORKSPACE}/${pomFilename}"
+    def pom = readMavenPom file: "${env.WORKSPACE}/${pomFilename}"
```

Comment:
Where does the `readMavenPom` come from? Is it the `jenkins-shared-libraries`?

Reply:
It comes from Pipeline Utility Steps, here https://www.jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#readmavenpom-read-a-maven-project-file

Reply:
Notice that I decided to use `readMavenPom` instead of executing `mvn help:evaluate -Dexpression=project.version -q -DforceStdout` because in this way I can avoid to clone the whole repository.. since only the main pom.xml file ios needed